### DISCUSSION TITLE
Slider not working properly

### DISCUSSION THREAD

**Discussion by mehrdadpirnazar** [artisan-roaster-scope/artisan#1549] [General]
Hello good time 
I'm running Artisan software on a touchscreen with Windows 11 But unfortunately the sliders do not work properly They seem to have a touch problem, but the screen is fine

**Comment by mehrdadpirnazar** [comment]
@MAKOMO

**Comment by mehrdadpirnazar** [comment]
@MAKOMO

**Comment by poundy** [comment]
Both your pings to Makomo were unwarranted. You should reconsider what you think this achieves - it certainly doesn't win you favours or bring more positive attention to your post. 

Your problem lacks significant detail. You have not told us what version or versions of Artisan you've tested sliders on. You've not confirmed that the slider works via mouse on the same computer. You've not told us whether this problem has just started happening, or if you are brand new to artisan and are just trying to get things working. You've not given us any information about the hardware that you have used and tested. You've not given us any information about how your Artisan setup is configured - you should attach your settings here (export a copy of them, rename the file to .txt, and then drag them into a reply on github, do not use email). You would get a lot further if you came here with more detail...

FWIW, I used to use the touch screen on my Win11 laptop for Artisan all the time, and both sliders and buttons worked fine. I rarely used the slider, but it did work through touch. This was across many many versions up to ~2.9 when my laptop was retired.

**Reply by mehrdadpirnazar** [reply]
Hello, thank you for your reply The sliders work properly with the mouse and there are no problems But they don't go up and down correctly on touch screens or they don't touch at all. I am using this touch screen https://www.waveshare.com/15.6inch-hdmi-lcd-h-with-case.htm This problem remains in Windows 10 and 11 and Linux versions Also, there is no problem in other software And this issue only arises in Slasher Artisan

**Reply by mehrdadpirnazar** [reply]
@poundy

**Reply by poundy** [reply]
I really take offence at how you've just pinged me here for no reason.  You must remember that the entire Artisan project is driven by volunteers and is not anyone's "paid job" and by incessantly pinging people like you've done for Makomo and now me, you detract from people's propensity to help...

The thread is clear - this is a localised issue for you and you alone, it's therefore your hardware that is causing your behaviour. Nobody else's hardware has even similar issues.  There's nothing we can do unless you're willing to donate a test rig that can reproduce the issue to someone to debug and examine.

**Reply by MAKOMO** [reply]
@poundy I am sorry for this users disturbing behavior. Maybe it is time to block or even report him. Let me know.

**Reply by poundy** [reply]
@MAKOMO it doesn't really worry me - I have all alerts turned on here so I can help others in the community who don't take liberties and understand that this is a community and don't have unrealistic expectations of volunteers... it's just that this person has repeatedly pinged you and now me directly for no apparent reason other than to "hurry us up", and hasn't had the courtesy to come back and acknowledge either what the troubleshooting has left them with, or apologise for the noise.

**Comment by Terracotta-6** [comment]
I'm using the touch screen on my Win11 & Ubuntu laptop dell computer to operate Artisan and the sliders and buttons work fine!

**Comment by roasterdave** [comment]
Your touch screen may need to be re-calibrated.  In Windows 11 open the `Control Panel>>  Hardware and Sound>> Tablet PC Settings`. Then click `Calibrate` and follow the directions.  Be precise with your finger when performing the calibration.

In Artisan you need to first touch the slider it to select it, then touch again on the slider handle and drag the handle to move it.

**Reply by mehrdadpirnazar** [reply]
Hello, thank you for your reply I calibrated and the problem was not solved Yes, the slider must be selected first and then move But unfortunately, you have to do this several times for the slider to move. Even I used touch pen for more accuracy But the problem still remains

**Reply by roasterdave** [reply]
This does not appear to be problem that can be solved in the Artisan code.  The signal is not getting through properly each time when you touch a slider handle.  Since the calibration did not resolve it, I recommend you take an alternate approach and use "relative" event buttons. Buttons present a larger touch target and should work with your system.  Relative event buttons allow plus or minus adjustment of the event setting by a fixed amount each time the button is touched.  The slider does not have to be visible for the button to work.  Here is a [blog post](https://artisan-roasterscope.blogspot.com/2015/10/increasing-heat.html) that explains them.
